integration-tests/src/dca.rs

```diff
@@ -588,6 +682,106 @@ mod omnipool {
 		});
 	}
 
+	#[test]
+	fn sell_schedule_execution_should_emit_swapped_event() {
+		TestNet::reset();
+		Hydra::execute_with(|| {
+			//Arrange
+			init_omnipool_with_oracle_for_block_10();
+			let alice_init_hdx_balance = 5000 * UNITS;
+			assert_ok!(Balances::force_set_balance(
+				RuntimeOrigin::root(),
+				ALICE.into(),
+				alice_init_hdx_balance,
+			));
+
+			let dca_budget = 1100 * UNITS;
+			let amount_to_sell = 100 * UNITS;
+			let schedule_id = 0;
+			let schedule1 =
+				schedule_fake_with_sell_order(ALICE, PoolType::Omnipool, dca_budget, HDX, DAI, amount_to_sell);
+			create_schedule(ALICE, schedule1);
+
+			//Act
+			run_to_block(11, 12);
+
+			//Assert
+			let swapped_events = get_last_swapped_events();
+			let last_two_swapped_events = &get_last_swapped_events()[swapped_events.len() - 2..];
+			pretty_assertions::assert_eq!(
+				last_two_swapped_events,
+				vec![
+					RuntimeEvent::AmmSupport(pallet_amm_support::Event::Swapped {
+						swapper: ALICE.into(),
+						filler: Omnipool::protocol_account(),
+						filler_type: pallet_amm_support::types::Filler::Omnipool,
+						operation: pallet_amm_support::types::TradeOperation::ExactIn,
+						inputs: vec![Asset::new(HDX, amount_to_sell)],
+						outputs: vec![Asset::new(LRNA, 49999999159957)],
+						fees: vec![Fee::new(LRNA, 24999999579, Omnipool::protocol_account()),],
+						operation_id: vec![
+							ExecutionType::DCA(schedule_id, 0),
+							ExecutionType::Router(1),
+							ExecutionType::Omnipool(2)
+						],
+					}),
+					RuntimeEvent::AmmSupport(pallet_amm_support::Event::Swapped {
+						swapper: ALICE.into(),
+						filler: Omnipool::protocol_account(),
+						filler_type: pallet_amm_support::types::Filler::Omnipool,
```

Comment:
Would be nice if this had more complex route since it's integration test but I guess we can do that separately.